acceptance/tests/wan-federation/wan_federation_test.go

```diff
@@ -190,11 +209,259 @@ func TestWANFederation(t *testing.T) {
 			}
 
 			logger.Log(t, "checking that connection is successful")
 			k8s.CheckStaticServerConnectionSuccessful(t, primaryHelper.KubectlOptsForApp(t), StaticClientName, "http://localhost:1234")
 		})
 	}
 }
+
+// Test failover scenarios with a static-server in dc1 and a static-server
+// in dc2. Use the static-client on dc1 to reach static-server on dc1 in the
+// nominal scenario, then cause a failure in dc1 static-server to see the static-client failover to
+// the static-server in dc2
+/*
+	dc1-static-client -- nominal -- > dc1-static-server in namespace ns1
+	dc1-static-client -- failover --> dc2-static-server in namespace ns1
+	dc1-static-client -- failover --> dc1-static-server in namespace ns2
+*/
+func TestWANFederationFailover(t *testing.T) {
+	cases := []struct {
+		name   string
+		secure bool
+	}{
+		{
+			name:   "secure",
+			secure: true,
+		},
+		{
+			name:   "default",
+			secure: false,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			env := suite.Environment()
+			cfg := suite.Config()
+
+			if cfg.EnableRestrictedPSAEnforcement {
+				t.Skip("This test case is not run with enable restricted PSA enforcement enabled")
+			}
+
+			primaryContext := env.DefaultContext(t)
+			secondaryContext := env.Context(t, 1)
+
+			primaryHelmValues := map[string]string{
+				"global.datacenter": primaryDatacenter,
+
+				"global.tls.enabled":   "true",
+				"global.tls.httpsOnly": strconv.FormatBool(c.secure),
+
+				"global.federation.enabled":                "true",
+				"global.federation.createFederationSecret": "true",
+
+				"global.acls.manageSystemACLs":       strconv.FormatBool(c.secure),
+				"global.acls.createReplicationToken": strconv.FormatBool(c.secure),
+
+				"connectInject.enabled":  "true",
+				"connectInject.replicas": "1",
+
+				"meshGateway.enabled":  "true",
+				"meshGateway.replicas": "1",
+
+				"global.enableConsulNamespaces":               "true",
+				"connectInject.consulNamespaces.mirroringK8S": "true",
+			}
+
+			if cfg.UseKind {
+				primaryHelmValues["meshGateway.service.type"] = "NodePort"
+				primaryHelmValues["meshGateway.service.nodePort"] = "30000"
+			}
+
+			releaseName := helpers.RandomName()
+
+			// Install the primary consul cluster in the default kubernetes context
+			primaryConsulCluster := consul.NewHelmCluster(t, primaryHelmValues, primaryContext, cfg, releaseName)
+			primaryConsulCluster.Create(t)
+
+			// Get the federation secret from the primary cluster and apply it to secondary cluster
+			federationSecretName := fmt.Sprintf("%s-consul-federation", releaseName)
+			logger.Logf(t, "Retrieving federation secret %s from the primary cluster and applying to the secondary", federationSecretName)
+			federationSecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(context.Background(), federationSecretName, metav1.GetOptions{})
+			require.NoError(t, err)
+			federationSecret.ResourceVersion = ""
```

Comment:
~ Is setting this to the empty string necessary for the copy to work? Wondering whether a comment or secret copy helper would be useful here

Reply:
Not sure, I copy/pasted it from the test above

Reply:
I'll create a copy helper to follow DRY a bit more